python/pyarrow/_flight.pyx

```diff
@@ -837,32 +842,42 @@ cdef class FlightMetadataWriter:
 
 
 cdef class FlightClient:
-    """A client to a Flight service."""
+    """A client to a Flight service.
+
+    Connect to a Flight service on the given host and port.
+
+    Parameters
+    ----------
+    location : Location
+        location to connect to
+    tls_root_certs : bytes or None
+        PEM-encoded
+    override_hostname : str or None
+        Override the hostname checked by TLS. Insecure, use with caution.
+    """
     cdef:
         unique_ptr[CFlightClient] client
 
-    def __init__(self):
-        raise TypeError("Do not call {}'s constructor directly, use "
-                        "`pyarrow.flight.FlightClient.connect` instead."
-                        .format(self.__class__.__name__))
-
-    @staticmethod
-    def connect(location, tls_root_certs=None, override_hostname=None):
-        """
-        Connect to a Flight service on the given host and port.
+    def __init__(self, location, tls_root_certs=None, override_hostname=None,
+                 wait_for_available=True, wait_timeout=5):
```

Comment:
i think we shouldn't multiply arguments to the constructor. The `wait_for_available` method can be called separately, it doesn't seem useful here.
(also, in the normal case, you probably don't need to wait when calling the constructor)

Reply:
I don't have a strong opinion here. However I don't prefer repeating arguments, blocking can be handy.

@wesm what is your preference?

Reply:
What I mean is that there is already a method for blocking. Also, this is done by calling a remote method on the server, and perhaps that remote method can be expensive in some cases (I doubt so in the common case, but still).

Reply:
I have additional concerns, see my comment below. I'm fine with removing it from the init.